ui: style fix of login password input

vladlos opened the pull request:
before: on login screen on Safari is displaying Yen symbols
insteas of dots
<img width="934" alt="Screenshot 2021-05-11 at 12 55 54" src="https://user-images.githubusercontent.com/12850886/117799357-ae89cf80-b25a-11eb-81e6-b94d2dd453aa.png">

after: fixed issue above and in addition fixed position of key
button overlap with "view password" button
<img width="959" alt="Screenshot 2021-05-11 at 13 00 44" src="https://user-images.githubusercontent.com/12850886/117799406-bb0e2800-b25a-11eb-9efa-3924828cf511.png">

Resolves: #49386

Release note (ui): style fix of password input field for Safari

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/64980)
<!-- Reviewable:end -->

koorosh commented on pkg/ui/src/components/input/input.styl:
```diff
@@ -61,3 +61,8 @@
       color $colors--functional-red-3
       position absolute
+  &__password
+    font-family sans-serif
```
Is it possible to use font styles from design system? (user can click "show password" and font styles with login input will be different)

elkmaster replied:
no, font from design system is the reason of this bug.

koorosh replied:
Then i would suggest to use the same font for "login" field as well just for consistency (In case these fonts visibly different). Other than that, it LGTM!

nathanstilwell replied:
I see what @koorosh is saying, but I think it would be more correct to have the password input component use this style (`font-style sans-serif`) when the input type is `password`, and inherit the font when the type is `text`.

I think the easiest way to do that would be to change this line in [`passwordInput.tsx`](https://github.com/cockroachdb/cockroach/blob/3c8736df144ecadfb717e99b35187c5678f88d47/pkg/ui/src/components/input/passwordInput.tsx#L66),

```tsx
const classes = cn(className, "crl-input", { "crl-input__password": !showPassword });
```

nathanstilwell reviewed (changes requested):
One suggestion for keeping the fonts consistent when the input types are `text`. It's a subtle difference, but I think an easy change.

vladlos committed:
ui: style fix of login password input

before: on login screen on Safari is displaying Yen symbols
insteas of dots
after: fixed issue above and in addition fixed position of key
button overlap with "view password" button

Resolves: #49386

Release note (ui): style fix of password input field for Safari

elkmaster commented:
sorry, misunderstood original comment, added font definition to password field with type="text"
cc: @nathanstilwell , @koorosh

nathanstilwell reviewed (approved):
LGTM

elkmaster commented:
bors r+

thtruo commented:
@elkmaster following up based on our offline convo. I'm still seeing Yen characters continue to show up in Safari DB Console login page
